src/shims/foreign_items.rs

```diff
@@ -163,6 +166,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                     .iter()
                     .find(|cnum| tcx.is_panic_runtime(**cnum))
                     .expect("No panic runtime found!");
+                check_arg_count(args, 1)?;
```

Comment:
THis is not needed: we go find the proper MIR of the function here and return it, which means the function calling code in the core engine will pass arguments and also validate that their number is right.